src/main/scala/com/lightbend/lagom/core/persistence/effects/core/persistence/AccountEntity.scala

```diff
@@ -52,9 +65,12 @@ class AccountEntity extends PersistentEntity[AccountCommand[_], AccountEvent, Ac
       }
 
   def withdrawCommandActions(account: Account) =
-    actions
+    actionBuilder
       .onCommand {
         Handler[Withdraw]
+          // IMS: I think we're causing a lot of extra effort here. I mean, because one case may throw an exception
+          // we have to (1) call `attempt` and (2) wrap each result into Success or Failure. It's a necessary
+          // evil, unfortunately :-(
```

Comment:
My example was not good, because it's too verbose. 
The main idea was to demonstrate that we can use different builders and vary on the types. 

That example could also be coded as
```
Handler[Withdraw]
  .persistOne {
      case cmd if account.amount - cmd.amount >= 0 =>WithdrawExecuted(cmd.amount)
      case cmd =>  throw new RuntimeException("Insufficient balance")
  }
```

The `.attempt` case is more interesting when we have something like:
```
Handler[Withdraw]
  .attempt.persistOne {
      case cmd  => account.validateWithdraw(cmd.amount).map(_ => WithdrawExecuted(cmd.amount))
  }
```
`validateWithdraw` would return a `Try`, but it could be something else. Like a user defined `Validation` type. The builder will need to be able to deal with if and build the write `Effect`

Reply:
I pushed a modified version: 
https://github.com/lagom/persistence-api-experiments/blob/bd2575be0fabb99753140725a4efbabf1369275e/src/main/scala/com/lightbend/lagom/core/persistence/effects/core/persistence/AccountEntity.scala#L54

Reply:
Wow, that's cool. New GitHub feature. Includes code fragment.